util/src/main/java/io/kubernetes/client/util/ConfigBuilder.java

```diff
@@ -0,0 +1,285 @@
+		return this;
+	}
+
+	public String getPassword() {
+		return password;
+	}
+
+	public ConfigBuilder setPassword(String password) {
+		this.password = password;
+		return this;
+	}
+
+	public String getApiKey() {
+		return apiKey;
+	}
+
+	public ConfigBuilder setApiKey(String apiKey) {
+		this.apiKey = apiKey;
+		return this;
+	}
+
+	public boolean isTrustCerts() {
+		return trustCerts;
+	}
+
+	public ConfigBuilder setTrustCerts(boolean trustCerts) {
+		this.trustCerts = trustCerts;
+		return this;
+	}
+
+	public String getbasePath() {
+		return basePath;
+	}
+
+	public ConfigBuilder setbasePath(String basePath) {
+		this.basePath = basePath;
+		return this;
+	}
+
+	public String getCertificateAuthorityFile() {
+		return certificateAuthorityFile;
+	}
+
+	public ConfigBuilder setCertificateAuthorityFile(String certificateAuthorityFile) {
+		this.certificateAuthorityFile = certificateAuthorityFile;
+		return this;
+
+	}
+
+	public String getCertificateAuthorityData() {
+		return certificateAuthorityData;
+	}
+
+	public ConfigBuilder setCertificateAuthorityData(String certificateAuthorityData) {
+		this.certificateAuthorityData = certificateAuthorityData;
+		return this;
+	}
+
+	public ConfigBuilder setClusterMode(boolean clusterMode) {
+		this.clusterMode = clusterMode;
+		return this;
+	}
+
+	public ConfigBuilder setKubeConfig(KubeConfig config) {
+		this.kubeConfig = config;
+		return this;
+	}
+
+	public ConfigBuilder setDefaultKubeConfigMode(boolean defaultKubeConfigMode) {
+		this.defaultKubeConfigMode = defaultKubeConfigMode;
+		return this;
+	}
+
+	public ConfigBuilder setKubeConfig(String fileName) throws FileNotFoundException {
+		this.kubeConfig = KubeConfig.loadKubeConfig(new FileReader(fileName));
+		return this;
+	}
+
+	public ConfigBuilder setKubeConfig(Reader input)  {
+		this.kubeConfig = KubeConfig.loadKubeConfig(input);
+		return this;
+	}
+
+	public ConfigBuilder setKubeConfig(InputStream stream)  {
+		this.kubeConfig = KubeConfig.loadKubeConfig(new InputStreamReader(stream));
+		return this;
+	}
+
+	public KeyManager[] getKeyMgrs() {
+		return keyMgrs;
+	}
+
+	public ConfigBuilder setKeyMgrs(KeyManager[] keyMgrs) {
+		this.keyMgrs = keyMgrs;
+		return this;
+	}
+
+	public boolean isVerifyingSsl() {
+		return verifyingSsl;
+	}
+
+	public ConfigBuilder setVerifyingSsl(boolean verifyingSsl) {
+		this.verifyingSsl = verifyingSsl;
+		return this;
+	}
+
+	public boolean isDefaultClientMode() {
+		return defaultClientMode;
+	}
+
+	public ConfigBuilder setDefaultClientMode(boolean defaultClientMode) {
+		this.defaultClientMode = defaultClientMode;
+		return this;
+	}
+
+	public String getApiKeyPrefix() {
+		return apiKeyPrefix;
+	}
+
+	public ConfigBuilder setApiKeyPrefix(String apiKeyPrefix) {
+		this.apiKeyPrefix = apiKeyPrefix;
+		return this;
+	}
+
+	public ApiClient build()  {
+
+		ApiClient client = new ApiClient();
+
+		if( kubeConfig !=null) {
+			client = Config.fromConfig(kubeConfig);
+			return client;
```

Comment:
I think this logic is a little odd.

I would expect that I could do:

```
client = new ConfigBuilder().setKubeConfig().setBasePath().build()
```

And it would mostly load the KubeConfig, but then override the basepath with the custom value I set.

Instead, it just returns the existing KubeConfig and ignores any other `setFoo` directives.

I think that this is going to confuse people, and doesn't match the desired use case.